### PR TITLE
fix(delegation): resolve 5 blocking and 8 non-blocking code review is…

### PR DESCRIPTION
**Pull request by tobyhede**
…sues

B1: Fix active delegation guard to allow re-delegation when childRunId is set
B2: Add --var-file flag to delegate command with proper precedence
B3: Add RD_CLAIM_TOKEN= marker in delegate output (human + JSON)
B4: Amend design doc to note RUNDOWN_RUN_ID removal
B5: Fix stale stepId in token scan — use contextSnapshot.step
N1: Use snake_case keys in delegate JSON output (token_hash, parent_run_id)
N2: Use DELEGATED uppercase and ASCII -> in human output
N3: Add truncateDelegationToken() helper (rdtk_XXX...YYYY format)
N4: Add claim --var-file integration test
N5: Add orphan reconciliation unit test
N6: Wrap claim command with withErrorHandling
N7: Use truncateDelegationToken in pipeline claimed output
N8: Add TOCTOU safety comment in delegation-lock

## Summary by CodeRabbit

* **New Features**
  * Added --var-file to delegate for loading/merging YAML variables
  * Claim output includes a visible claim marker and consistent truncated token display
  * CLI JSON outputs now use snake_case keys

* **Bug Fixes**
  * Re-delegation allowed when prior delegation is inactive
  * Text output shows DELEGATED status consistently

* **Refactor**
  * Claim command error handling unified for clearer end-user output

* **Tests**
  * Expanded delegation, locking, scanning, context and edge-case coverage

### CODE REVIEW

File: packages/core/__tests__/runbook/delegation-context.test.ts [hunk 1]
```diff
@@ -200,4 +200,180 @@ describe('reconstituteContextVars', () => {
...
+    };
+
+    const originalVars = { ...snapshot.vars };
+    const originalAncestors = [...snapshot.ancestors];
+
+    reconstituteContextVars(snapshot);
+
+    // Verify snapshot wasn't mutated
+    expect(snapshot.vars).toEqual(originalVars);
+    expect(snapshot.ancestors).toEqual(originalAncestors);
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-context.test.ts:329]: Addressed — the immutability test was removed during the test rewrite in commit 38a4526. The current test file (204 lines) no longer contains this assertion pattern.

File: packages/core/__tests__/runbook/delegation-lock.test.ts [hunk 2]
```diff
@@ -136,4 +136,129 @@ describe('DelegationLock', () => {
...
+    await expect(fs.access(lockA)).resolves.toBeUndefined();
+    await expect(fs.access(lockB)).resolves.toBeUndefined();
+    await expect(fs.access(lockC)).resolves.toBeUndefined();
+
+    // Release all
+    await lock.release('run-a');
+    await lock.release('run-b');
+    await lock.release('run-c');
+  });
+});
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-lock.test.ts]: Addressed — formatting passes now (`npm run format:check` succeeds). The file was also significantly rewritten (now 140 lines vs 264+ referenced).

File: packages/core/__tests__/runbook/delegation-lock.test.ts [hunk 3]
```diff
@@ -136,4 +136,129 @@ describe('DelegationLock', () => {
...
+
+    // Second lock should retry and eventually acquire
+    const lock2 = new DelegationLock(tmpDir);
+    await lock2.acquire('run-concurrent');
+
+    // Verify lock2 now owns the lock
+    const lockPath = path.join(tmpDir, '.claude/rundown/locks/run-run-concurrent.delegation.lock');
+    const content = JSON.parse(await fs.readFile(lockPath, 'utf8'));
+    expect(content.pid).toBe(process.pid);
+
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-lock.test.ts:156]: Addressed — the concurrent retry test was removed during the test rewrite in commit 38a4526. The current file (140 lines) no longer contains this test.

File: packages/core/__tests__/runbook/delegation-lock.test.ts [hunk 4]
```diff
@@ -136,4 +136,129 @@ describe('DelegationLock', () => {
...
+    await fs.writeFile(lockPath, JSON.stringify(oldLock));
+
+    // Should reclaim the old lock
+    await lock.acquire('run-old');
+
+    const content = JSON.parse(await fs.readFile(lockPath, 'utf8'));
+    expect(content.pid).toBe(process.pid);
+    // Timestamp should be recent, not the old one
+    const age = Date.now() - new Date(content.created_at).getTime();
+    expect(age).toBeLessThan(1000);
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-lock.test.ts:179]: Addressed — the stale threshold test with the brittle time bound was removed during the test rewrite in commit 38a4526.

File: packages/core/__tests__/runbook/delegation-lock.test.ts [hunk 5]
```diff
@@ -136,4 +136,129 @@ describe('DelegationLock', () => {
...
+
+    // Start acquire, then delete the lock file to simulate race
+    const acquirePromise = lock.acquire('run-vanish');
+
+    // Wait briefly for the first retry to hit
+    await new Promise((resolve) => setTimeout(resolve, 80));
+
+    // Remove lock file (simulates another process releasing)
+    await fs.unlink(lockPath).catch(() => { /* ignore if already gone */ });
+
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-lock.test.ts:206]: Addressed — the race-condition test with sleep-based orchestration was removed during the test rewrite in commit 38a4526.

File: packages/core/__tests__/runbook/delegation-scan.test.ts [hunk 6]
```diff
@@ -137,5 +175,170 @@ describe('DelegationScanService', () => {
...
+
+      const state = makeState('run-large-substeps', {
+        substepStates: substeps,
+      });
+      await writeState(state);
+
+      const result = await scanner.findByToken(token);
+      expect(result).not.toBeNull();
+      expect(result!.substepId).toBe('50');
+    });
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-scan.test.ts:346]: Fixed in commit 3e02224 — added explicit type annotation for the substeps array to include the `delegation` property.

File: packages/core/__tests__/runbook/delegation-service.test.ts [hunk 7]
```diff
@@ -291,4 +316,192 @@ describe('createDelegation', () => {
...
+      contextSnapshot: { vars: {}, ancestors: [] },
+      childRunId: 'completed-run-123',
+      createdAt: '2026-02-27T10:00:00.000Z',
+      cancelledAt: null,
+    };
+    const state = makeState({
+      substepStates: [
+        { id: '1', status: 'completed', delegation: completedDelegation },
+        { id: '2', status: 'pending' },
+      ],
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-service.test.ts:361]: Fixed in commit 3e02224 — changed `'completed'` to `'done'` to match the SubstepState type.

File: packages/core/__tests__/runbook/delegation-service.test.ts [hunk 8]
```diff
@@ -291,4 +316,192 @@ describe('createDelegation', () => {
...
+    const result = createDelegation({ state, stepId: '1.2', childRunbookPath: 'child.md' }, steps);
+
+    const ss1 = result.updatedSubstepStates.find((ss) => ss.id === '1');
+    expect(ss1?.agentId).toBe('agent-1');
+    expect(ss1?.startedAt).toBe('2026-02-27T10:00:00.000Z');
+    expect(ss1?.delegation).toBeUndefined();
+
+    const ss2 = result.updatedSubstepStates.find((ss) => ss.id === '2');
+    expect(ss2?.delegation).toBeDefined();
+  });
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/__tests__/runbook/delegation-service.test.ts:413]: Fixed in commit 3e02224 — removed non-existent `startedAt` property, replaced with `status` assertion.
